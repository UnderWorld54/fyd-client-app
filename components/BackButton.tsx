import { Ionicons } from '@expo/vector-icons';
import { useRouter } from 'expo-router';
import { StyleSheet, TouchableOpacity } from 'react-native';

export default function BackButton() {
  const router = useRouter();
  return (
    <TouchableOpacity style={styles.button} onPress={() => router.back()}>
      <Ionicons name="chevron-back" size={28} color="#222" />
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  button: {
    backgroundColor: '#f5f7fa',
    borderRadius: 16,
    padding: 8,
    alignSelf: 'center',
  },
}); 