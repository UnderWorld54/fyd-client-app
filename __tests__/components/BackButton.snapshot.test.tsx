import BackButton from '@/components/BackButton';
import { render } from '@testing-library/react-native';
import { router } from 'expo-router';
import React from 'react';

describe('BackButton Component Snapshots', () => {
  it('matches snapshot when canGoBack is true', () => {
    // Mock canGoBack to return true
    (router.canGoBack as jest.Mock).mockReturnValue(true);
    
    const tree = render(<BackButton />).toJSON();
    
    expect(tree).toMatchSnapshot();
  });

  it('matches snapshot when canGoBack is false', () => {
    // Mock canGoBack to return false
    (router.canGoBack as jest.Mock).mockReturnValue(false);
    
    const tree = render(<BackButton />).toJSON();
    
    expect(tree).toMatchSnapshot();
  });
});