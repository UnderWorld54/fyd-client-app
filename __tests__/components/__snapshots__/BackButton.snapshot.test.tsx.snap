// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackButton Component Snapshots matches snapshot when canGoBack is false 1`] = `
<View
  style={
    {
      "width": 40,
    }
  }
/>
`;

exports[`BackButton Component Snapshots matches snapshot when canGoBack is true 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "height": 40,
      "justifyContent": "center",
      "opacity": 1,
      "width": 40,
    }
  }
>
  <Ionicons
    color="#000"
    name="chevron-back"
    size={24}
  />
</View>
`;
